render/loadsheet.go

```diff
@@ -2,93 +2,67 @@ package render
 
 import (
 	"image"
+	"path/filepath"
 
+	"github.com/oakmound/oak/v3/alg/intgeom"
 	"github.com/oakmound/oak/v3/oakerr"
 )
 
-// LoadSprites calls LoadSheet and then Sheet.ToSprites.
-func LoadSprites(directory, fileName string, w, h, pad int) ([][]*Sprite, error) {
-	sh, err := LoadSheet(directory, fileName, w, h, pad)
-	if sh != nil {
-		return sh.ToSprites(), err
-	}
-	return nil, err
-}
-
-// LoadSheet loads a file in some directory with sheets of (w,h) sized sprites,
-// where there is pad pixels of vertical/horizontal empty space between each sprite.
+// LoadSheet loads a file in some directory with sheets of (w,h) sized sprites.
 // This will blow away any cached sheet with the same fileName.
-func LoadSheet(directory, fileName string, w, h, pad int) (*Sheet, error) {
-
+func (c *Cache) LoadSheet(file string, cellSize intgeom.Point2) (*Sheet, error) {
 	var rgba *image.RGBA
 	var ok bool
 	var err error
 
-	imageLock.RLock()
-	rgba, ok = loadedImages[fileName]
-	imageLock.RUnlock()
-
 	if !ok {
-		rgba, err = loadSprite(directory, fileName, 0)
+		rgba, err = c.loadSprite(file, 0)
 		if err != nil {
 			return nil, err
 		}
 	}
 
-	sheet, err := MakeSheet(rgba, w, h, pad)
+	sheet, err := MakeSheet(rgba, cellSize)
 	if err != nil {
 		return nil, err
 	}
 
-	sheetLock.Lock()
-	defer sheetLock.Unlock()
-	loadedSheets[fileName] = sheet
+	c.sheetLock.Lock()
```

Comment:
Given that we no longer support oddly padded sprite sheets (which is probably for the best) maybe we pull out the loadedsheets logic so that implementers can choose to set up their own parsing strategy for some sheets. For example lets say for whatever reason there is a desire to work with a sheet that has had padding or something else odd applied to it from an artistic source and the implementers want to keep the file unchanged at rest but still want to load it into oak. 
In that case it would be nice if they could set up a flow where they could custom roll a function like loadsheet except it performs some mutations.
I know this would make the cache interface a bit more annoying as it would also need to expose the persistence part but it may be a nice addition.

Reply:
For custom sheet formats (which definitely exist, there's a lot of sheets which aggressively pad any number of different sized elements into the same rectangle), I don't think we could assume that our sheet type would properly represent them. The thing to do, I think, would be to add hooks into the batch loading process that would let you add your own caching layer that ran on images as we loaded them. 

However, I don't know if there's an ask for this, and it likely would be a worse experience than loading and caching the files yourself, or letting us load the files and then doing a manual pass to convert known sheets into known formats following.